src/bentoml/_internal/frameworks/diffusers.py

```diff
@@ -0,0 +1,355 @@
+                )
+
+            src_dir = snapshot_download(
+                model_name_or_path,
+                proxies=proxies,
+                revision=revision,
+            )
+
+        model_config_file = os.path.join(src_dir, DIFFUSION_MODEL_CONFIG_FILE)
+        if not os.path.exists(model_config_file):
+            raise BentoMLException(f'artifact "{src_dir}" is not a Diffusion model')
+
+        shutil.copytree(src_dir, diffusion_model_dir, symlinks=False, ignore=ignore)
+
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+
+    partial_kwargs: t.Dict[str, t.Any] = bento_model.info.options.partial_kwargs  # type: ignore
+    pipeline_class: type[diffusers.DiffusionPipeline] = (
+        bento_model.info.options.pipeline_class or diffusers.StableDiffusionPipeline
+    )
+    scheduler_class: type[
+        diffusers.SchedulerMixin
+    ] | None = bento_model.info.options.scheduler_class
+    custom_pipeline: str | None = bento_model.info.options.custom_pipeline
+    _enable_xformers: str | None = bento_model.info.options.enable_xformers
+    _torch_dtype: str | torch.dtype | None = bento_model.info.options.torch_dtype
+
+    class DiffusersRunnable(bentoml.Runnable):
+        SUPPORTED_RESOURCES = ("nvidia.com/gpu", "cpu")
+        SUPPORTS_CPU_MULTI_THREADING = True
+
+        def __init__(self):
+            super().__init__()
+
+            if torch.cuda.is_available() and _torch_dtype is None:
+                torch_dtype = torch.float16
+            else:
+                torch_dtype = _torch_dtype
+
+            enable_xformers: bool = False
+            if torch.cuda.is_available() and _enable_xformers is None:
+                if is_xformers_available():
+                    enable_xformers: bool = True
+
+            self.pipeline: diffusers.DiffusionPipeline = load_model(
+                bento_model,
+                pipeline_class=pipeline_class,
+                scheduler_class=scheduler_class,
+                torch_dtype=torch_dtype,
+                custom_pipeline=custom_pipeline,
+                enable_xformers=enable_xformers,
+            )
+
+            if torch.cuda.is_available():
+                self.pipeline.to("cuda")
+
+    def make_run_method(
+        method_name: str, partial_kwargs: dict[str, t.Any] | None
+    ) -> t.Callable[..., t.Any]:
+        def _run_method(
+            runnable_self: DiffusersRunnable,
+            *args: t.Any,
+            **kwargs: t.Any,
+        ) -> t.Any:
+
+            if method_partial_kwargs is not None:
+                kwargs = dict(method_partial_kwargs, **kwargs)
+
+            raw_method = getattr(runnable_self.pipeline, method_name)
+            res = raw_method(*args, return_dict=False, **kwargs)
```

Comment:
Shouldn't we allow users to set this instead of ourselves? it might be different from what users might expects.

Reply:
@aarnphm do you mean `return_dict`? If set to `True` then we will have some issue to pickling the output. Have you met similar option in transformers implementation?

Reply:
I think I will set  default to False but add an option to let user change it if they really want to

Reply:
I think that if they are passing it into kwargs it will fail right? best we don't do it here and let the framework do it. If users want to change that they can do that themselves.

Reply:
I have slightly change this part so user can change the value.

BTW all DiffusionPipeline should accept `return_dict` so the calling will never fail